{
  try {
    fun.run();
  }
 catch (  MetricsException e) {
    assertTrue("expected exception",e.getMessage().startsWith(prefix));
    return;
  }
  fail("should've thrown '" + prefix + "...'");
}
